xds: import rbac.proto from Envoy's repository

cindyxue opened the pull request:
This PR does the following thing:

Added dependencies needed for `rbac.proto`

@ZhenLian

sanjaypujare commented:
There is a conflict in the file. 

Secondly the imported files should also be part of this PR. The import is not run as part of the build.

cindyxue commented:
@sanjaypujare Hi Sanjay, do you mean I need to `git add` files generated by running `import.sh` too? Thanks!

cindyxue commented:
@sanjaypujare Ok. Thanks for the clarification!

cindyxue commented:
@voidzcy After I moved `googleapis` deps to a separate folder, I'm getting this error ```Import "google/api/expr/v1alpha1/syntax.proto" was not found or had errors``` when running `./gradlew build`

I think it uses `grpc-java/xds/third_party/envoy/src/main/proto/google/api/expr/v1alpha1/syntax.proto` as the default path, which is why it couldn't find the file after we changed the path to `xds/third_party/googleapis/src/main/proto/google/api/expr/v1alpha1/syntax.proto`. Is there a way to fix it?

voidzcy commented:
You need to add `third_party/googleapis/src/main/proto` into the `proto` sourceDirectorySet of the `main` sourceSet in buildscript. Otherwise, the protobuf-gradle-plugin is not able to find the source of googleapis protos you just added. You can see how the buildscript does that [here](https://github.com/grpc/grpc-java/blob/c3ea8e7c91c54d9af7d1086336812922527271a7/xds/build.gradle#L65-L69).

Btw, don't forget to shade googleapis (See [how shade is done](https://github.com/grpc/grpc-java/blob/c3ea8e7c91c54d9af7d1086336812922527271a7/xds/build.gradle#L86-L96) and [how shading works in gradle](https://imperceptiblethoughts.com/shadow/introduction/)).

cindyxue commented:
@voidzcy Ok, thank you!

cindyxue commented:
@voidzcy Hi, I fixed the `googleapis` directory and also added the import files. Could you please check it again? Thank you so much :)

voidzcy commented:
Why there are so many submodules added? We shouldn't have any. Did you do anything wrong when checking in changes?

cindyxue commented:
I just did `git add .`. Is there any command I need to run to get rid of the submodules before I push?

voidzcy commented on xds/third_party/envoy/import.sh:
```diff
@@ -135,4 +140,4 @@ do
 popd
 
-popd
+popd
```
Did you deleted the newline at the end of the file?

cindyxue replied:
Oops, my bad. I'll change it back!

voidzcy commented on xds/third_party/googleapis/import.sh:
```diff
@@ -0,0 +1,56 @@
+done
+popd
+
+popd
```
New line at the end of the file.

cindyxue replied:
Ok!

voidzcy commented on xds/third_party/googleapis/import.sh:
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Copyright 2018 The gRPC Authors
```
2020

cindyxue replied:
Thanks for the catch!

ZhenLian commented on xds/build.gradle:
```diff
@@ -66,6 +66,7 @@ sourceSets {
             srcDir 'third_party/envoy/src/main/proto'
             srcDir 'third_party/protoc-gen-validate/src/main/proto'
             srcDir 'third_party/udpa/src/main/proto'
+            srcDir 'third_party/googleapis/src/main/proto'
```
@voidzcy Shall we keep the alphabetical order of this?

voidzcy replied:
Doesn't matter.

ZhenLian replied:
OK, SG!

cindyxue commented:
@voidzcy @ZhenLian 
Fixed some small issues! Feel free to let me know if there are more changes to make. Thanks again :D

ZhenLian commented on xds/third_party/googleapis/import.sh:
```diff
@@ -0,0 +1,56 @@
+
+pushd `git rev-parse --show-toplevel`/xds/third_party/googleapis
+
+# clone the envoy github repo in a tmp directory
```
nit: s/envoy/googleapis

cindyxue replied:
Do you mean I need to change `# clone the envoy github repo in a tmp directory` to `# clone the s/envoy/googleapis github repo in a tmp directory` here?

ZhenLian replied:
I meant to replace "envoy" with "googleapis" since this is cloning the googleapis repo.
`# clone the googleapis github repo in a tmp directory` should be fine.

cindyxue replied:
Got it! Thanks for the clarification :)
